midas/codes/nuscale_lut.py

```diff
@@ -0,0 +1,71 @@
+import h5py
+import os
+import numpy as np
+import logging
+
+def evaluate(soln, input): #!TODO: Put parameters in docstring
+    """
+    This function will find the output parameters for the NuScale SMR based on the SMR database of loading pattern calculations
+    and write them into a dictionary in the solution.parameters object
+
+    Written by Cole Howard. 10/29/2024
+    """
+    #Each objective is stored as one index in a single array within the hdf5 file, so I am getting each specific value
+    objectives, BU = read_hdf5(soln.chromosome) #TODO!: Add cost back in
+    soln.parameters["cycle_length"]["value"] = objectives[0]
+    soln.parameters["fdeltah"]["value"] = objectives[1]
+    soln.parameters["pinpowerpeaking"]["value"] = objectives[2]
+    soln.parameters["max_boron"]["value"] = objectives[3]
+    #solution.parameters['cycle_cost'] = cost
+
+    # Adding in burnup parameters in case it is used later on
+    #soln.parameters["max_burnup"]["value"] = max(BU) #TODO!: Add burnup parameters to solution object
+    #soln.parameters["min_burnup"]["value"] = min(BU)
+    #soln.parameters["average_burnup"]["value"] = np.mean(BU)
+
+    return soln
+
+def read_hdf5(soln):
+    """
+    This function will search the hdf5 files containing the NuScale SMR database of loading patterns for the user-defined loading pattern,
+    then return the parameters for that solution
+
+    Parameters:
+        individual: array
+            array containing loading pattern for NuScale SMR
+
+    Written by Cole Howard. 10/29/2024
+    """
+    individual = []
+    for sol in soln:
+        if sol == 'NSFA23':
+            individual.append(int(2))
+        elif sol == 'NSFA23GAD':
+            individual.append(int(3))
+        elif sol == 'NSFA39':
+            individual.append(int(4))
+        elif sol == 'NSFA39GAD':
+            individual.append(int(5))
+        elif sol == 'NSFA455':
+            individual.append(int(6))
+        elif sol == 'NSFA455GAD':
```

Comment:
where do you require that the user provide these input options? what if a different option is provided? I think you might need additional logic in the input_parser.py to enforce this.

Reply:
I just pushed a new version of this into my fork, I believe you should see it in the current pull request but I can create a new one if need be. I changed the logic so the fuel assembly name has no bearing on the optimization itself, and takes the "type" parameter under fuel in the assembly options for the corresponding fuel assembly name. I also added a check to make sure it's a valid type (i.e. between 2-7), but can move this logic to input_parser or somewhere else if need be. Let me know if anything else here needs to be changed

Reply:
I can certainly change this if you'd like me to. However, this should work with any conceivable name a user might input for the fuel assembly name, with my most recent push. It would print the chromosomes as the user-defined names (in this case something like NSFA455), and this code interface would look into the genome/fa options and match it with the corresponding fuel type (2-7) regardless of how the user defined it. I believe validate_input already checks to make sure that each fuel assembly in decision variables has a matching fuel parameter under fa options, so this is likely to work as is. A quick question, I saw that my changes were approved, is there something I need to do on my end to merge the branches, or are you waiting for me to address these additional comments before merging them?

Reply:
I approved the pull request because I didn't think my additional comments were vital, but I didn't merge the request in case you wanted to make any final changes based on my comments. Once approval has been given, you should be able to initiate the merge at your convenience.

Reply:
Understood. I'd like to go ahead and merge it now and address some of these comments later, so that others can use the database if they'd like. However, I do not have write access, so even with the approved pull request I believe I am unable to merge these myself. Would it be possible for you to merge these branches now and then I can come back to these comments later?